src/blueapi/service/model.py

```diff
@@ -54,10 +54,15 @@ class PlanModel(BlueapiBaseModel):
     """
 
     name: str = Field(description="Name of the plan")
+    properties: Dict[str, Any] = Field(description="Plan arguments description")
+    required: List[str] = Field(description="Names of required arguments")
 
     @classmethod
     def from_plan(cls, plan: Plan) -> "PlanModel":
-        return cls(name=plan.name)
+        schema = plan.model.schema()
+        properties = schema.get("properties") or {}
+        required = schema.get("required") or []
+        return cls(name=plan.name, properties=properties, required=required)
```

Comment:
To give some context: I had always hoped to unify `Plan` and `PlanModel`. The problem was that you could easily serialize `plan.schema` the way you've done it here but not so easily deserialize it back to a pydantic model. So the only options were to make it un-deserializable, make a new model without a schema model in it, or move the schema model somewhere else.

Don't think this is any more soluble than before, but just in case, got any ideas?

Reply:
Decided in meeting to use JSONSchema representation of the arguments under the `schema` key at the same level as the `name` key